### PR TITLE
Emerging Rainbow

### PR DESCRIPTION
**Pull request by i-shreyagarg**
# Emerging Rainbow in C++

This code is about making a rainbow emerge from time=0 seconds. This code helps to gain a deeper understanding of using graphics in C++.

## Type of change

-  New feature (non-breaking change which adds functionality)

# Checklist:

-  My code follows the style guidelines of this project
-  I have performed a self-review of my own code
-  I have commented my code, particularly in hard-to-understand areas
-  I have made corresponding changes to the documentation
-  My changes generates no new warnings
-  I have added tests that prove my fix is effective or that my feature works
-  New and existing unit tests pass locally with my changes
-  Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: snippets/c++/Emerging_Rainbow.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+return 0;
+}
+```
+#### Output
+https://drive.google.com/file/d/1BMZbuWzrROUu0wjm0eleGJzrJbUcl7fH/view?usp=sharing
+#### Contributed By
+| Name | Github Username | College Name|
+| --- | --- | --- |
+| Shreya Garg | einhawking | Maharahja Agrasen Institute of Technology |
+
```
Comment (kaustubhgupta) [inline snippets/c++/Emerging_Rainbow.md]: Remove these trailing spaces

File: snippets/c++/Emerging_Rainbow.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	{
+		delay(100);
+		setcolor(i/10);
+		arc(x,y,0,180,i-10);
+	}
+return 0;
+}
+```
+#### Output
+https://drive.google.com/file/d/1BMZbuWzrROUu0wjm0eleGJzrJbUcl7fH/view?usp=sharing
```
Comment (kaustubhgupta) [inline snippets/c++/Emerging_Rainbow.md]: Insert the output image here only

File: snippets/c++/Emerging_Rainbow.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+#### Contributed By
+| Name | Github Username | College Name|
+| --- | --- | --- |
```
Comment (kaustubhgupta) [inline snippets/c++/Emerging_Rainbow.md]: Correct the spelling of our college

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by i-shreyagarg** [comment]
I've made the required changes, please check.

**Review by kaustubhgupta — APPROVED** [review]

File: snippets/c++/Emerging_Rainbow/Emerging_Rainbow.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Emerging Rainbow
+This code is about making a rainbow emerge from time=0 seconds.
+#### Langauge
```
Comment (sakshamarora1) [inline snippets/c++/Emerging_Rainbow/Emerging_Rainbow.md]: We should keep the heading size fixed to h2 (i.e. ##) just like in other files.

File: snippets/c++/Emerging_Rainbow/Emerging_Rainbow.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+C++
+#### Type of Issue
+Feature
```
Comment (sakshamarora1) [inline snippets/c++/Emerging_Rainbow/Emerging_Rainbow.md]: We don't need to add a type of issue heading here.
```suggestion
```

File: snippets/c++/Emerging_Rainbow/Emerging_Rainbow.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+# Emerging Rainbow
+This code is about making a rainbow emerge from time=0 seconds.
```
Comment (sakshamarora1) [inline snippets/c++/Emerging_Rainbow/Emerging_Rainbow.md:3]: Adding this under a description sub heading will be better.

File: snippets/c++/Emerging_Rainbow/Emerging_Rainbow.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+	{
+		delay(100);
+		setcolor(i/10);
+		arc(x,y,0,180,i-10);
+	}
+return 0;
+}
+```
+#### Output
+![Rainbow Image](https://github.com/einhawking/cool_code_snippets/blob/master/snippets/c%2B%2B/Emerging_Rainbow/Rainbow_image.PNG)
```
Comment (sakshamarora1) [inline snippets/c++/Emerging_Rainbow/Emerging_Rainbow.md]: If possible can you add a gif of the output, it'll look very cool. :D

**Review by sakshamarora1 — CHANGES_REQUESTED** [review]
@einhawking This looks great! Thanks for this contribution. Just a few changes and it'll be ready to merge!

**Comment by i-shreyagarg** [comment]
I've left the speed of gif as the original speed. Hope it's okay.

**Pull request merged**